app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -116,6 +125,13 @@ def get_instance_hardware(instance, flavor)
     end
   end
 
+  def get_instance_labels(instance, tags)
```

Comment:
seems like 1 method get_vm_or_template_labels should be enough

and just pass the 2 lazy_finds as args